context.go

```diff
@@ -181,6 +181,10 @@ func (c *Ctx) resolveProjectRoot(path string) (string, error) {
 func (c *Ctx) SplitAbsoluteProjectRoot(path string) (string, error) {
 	srcprefix := filepath.Join(c.GOPATH, "src") + string(filepath.Separator)
 	if internal.HasFilepathPrefix(path, srcprefix) {
+		if len(path) <= len(srcprefix) {
+			return "", errors.New("projects directly within $GOPATH/src are not supported currently")
```

Comment:
> I changed the error message to: "projects directly within $GOPATH/src are not supported currently"
I choose to not specify initializing since this error could occur in when running `dep remove`.

Ahh, I'm sorry, I wasn't understanding the issue with my suggested wording until now. My phrasing suggested that this was only about initializing a project, but the wording needs to be equally applicable for other `dep` subcommands.

Here's a revised suggestion:

`"dep does not currently support using $GOPATH/src as the project root."`